app/(gcforms)/[locale]/(form administration)/form-builder/[id]/publish/PrePublishDialog.tsx

```diff
@@ -0,0 +1,179 @@
+import { useTranslation } from "@i18n/client";
+import { Button, Alert } from "@clientComponents/globals";
+import { Dialog, useDialogRef, Radio, TextArea } from "@formBuilder/components/shared";
+import React, { useState } from "react";
+import Select, { SingleValue } from "react-select";
+
+export const PrePublishDialog = ({
+  handleClose,
+  handleConfirm,
+  setFormType,
+  setDescription,
+  setReasonForPublish,
+  reasonForPublish,
+  formType,
+  description,
+}: {
+  handleClose: () => void;
+  handleConfirm: () => void;
+  setFormType: (formType: string) => void;
+  setDescription: (description: string) => void;
+  setReasonForPublish: (reasonForPublish: string) => void;
+  reasonForPublish: string;
+  formType: string;
+  description: string;
+}) => {
+  const { t } = useTranslation("form-builder");
+  const dialog = useDialogRef();
+
+  const [prePublishStep, setPrePublishStep] = useState(0);
```

Comment:
A minor suggestion to use an enum to help describe the two levels.

Reply:
Changed!~ :)